web/views/analyses.py

```diff
@@ -72,7 +72,7 @@ def index(self):
         page = int(request.args.get('page', 1))
 
         analyses = current_user.analyses.find().sort('_id', DESCENDING).limit(PER_PAGE).skip((page - 1) * PER_PAGE)
-        pagination = Pagination(page=page, per_page=PER_PAGE, total=analyses.count(), css_framework='bootstrap3')
+        pagination = Pagination(page=page, per_page=PER_PAGE, total=len(list(analyses.clone())), css_framework='bootstrap3')
```

Comment:
In my understanding this is not the same thing.

- `.count()` will ignore the limit and return the count of all results
- Your proposal will not ignore the limit and will perform the operation twice (once for the count and once to actually get the analyses)

The same goes for all `count` replacements.

`count_documents` appears to be the correct replacement, but not as easy to use. We could also chose to change how pagination is handled because having a count is not that useful anyway.

Reply:
Hum...
Thanks for bringing this to me, I missed the limit part.

I agree this code is a bit dirty: I was trying to avoid changing how the pagination is done entirely, and as you said `count_documents()` is not as easy to use. I will rework this and come back to you with a new proposal